content/docs/04-clusters/03-edge/04-edgeforge-workflow/03.5-palette-canvos.md

```diff
@@ -0,0 +1,318 @@
+      description: "Instructs the installer to turn the host machine off once the installation is complete.",
+    },
+    {
+      x: 190,
+      y: 300,
+      label: 3,
+      description: "Sets the login credentials for Edge hosts. The login credentials will allow you to SSH log into the edge host for debugging purposes.",
+      tooltipPlacement: "rightTop",
+    },
+  ]}
+>
+
+
+```shell
+cat <<'EOF' > user-data
+#cloud-config
+stylus:
+  site:
+    edgeHostToken: aUAxxxxxxxxx0ChYCrO
+install:
+  poweroff: true
+users:
+  - name: kairos
+    passwd: kairos
+EOF
+```
+
+</PointsOfInterest>
+
+## Build Edge Native Artifacts
+Issue the following command to build the artifacts. 
+<br />
+
+```shell
+./earthly.sh +build-all-images --PE_VERSION=$(git describe --abbrev=0 --tags)
+```
+
+```bash coloredLines=2-2|#006622
+# Output condensed for readability
+===================== Earthly Build SUCCESS ===================== 
+Share your logs with an Earthly account (experimental)! Register for one at https://ci.earthly.dev.
+```
+This command will take up to 15-20 minutes to finish.
+<br />
+
+# Validate
+List the edge installer ISO image and checksum by issuing the following command from the **CanvOS** directory.
+<br />
+
+```shell
+ls build/
+```
+
+```shell coloredLines=2-3|#666666
+# Output
+palette-edge-installer.iso      
+palette-edge-installer.iso.sha256
+```
+You can provision as many Edge hosts using this ISO image as you desire.
+
+List the Docker images to show two provider OS images, one compatible with lightweight Kubernetes (K3s) v1.24.6 and another with K3s v1.25.2.
+<br />
+
+```shell
+docker images
+```
+
+```bash coloredLines=3-4|#666666
+# Output
+REPOSITORY        TAG                       IMAGE ID        CREATED         SIZE
+ttl.sh/ubuntu     k3s-1.25.2-v3.3.3-demo    b3c4956ccc0a    6 minutes ago   2.49GB
+ttl.sh/ubuntu     k3s-1.24.6-v3.3.3-demo    fe1486da25df    6 minutes ago   2.49GB
+earthly/earthly   v0.7.4                    d771cc8edc38     2 weeks ago    333MB
+```
+
+Suppose you want to use these provider images in your cluster profile. In that case, you can push them to the *ttl.sh* or any other image registry and add the following custom content within the `options` attribute of the **OS layer** of your cluster profile.    
+<br />
+
+```yaml
+  system.uri: "{{ .spectro.pack.edge-native-byoi.options.system.registry }}/{{ .spectro.pack.edge-native-byoi.options.system.repo }}:{{ .spectro.pack.edge-native-byoi.options.system.k8sDistribution }}-{{ .spectro.system.kubernetes.version }}-{{ .spectro.pack.edge-native-byoi.options.system.peVersion }}-{{ .spectro.pack.edge-native-byoi.options.system.customTag }}"
+  system.registry: ttl.sh
+  system.repo: ubuntu
+  system.k8sDistribution: k3s
+  system.osName: ubuntu
+  system.peVersion: v3.3.3
+  system.customTag: demo
+  system.osVersion: 22
+```
+You must verify and change these attributes' values, as applicable to you, before using them in your cluster profile. 
+<br />
```

Comment:
We can link to the [Model Edge Native Cluster Profile](https://docs.spectrocloud.com/clusters/edge/site-deployment/model-profile) guide. The challenge with this validation is that the user would have to deploy an Edge host, deploy an Edge cluster to validate this. 

Instead, let's focus on ensuring the artifact is created and available in the registry.